composer/utils/file_helpers.py

```diff
@@ -0,0 +1,305 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Utility for retrieving files."""
+
+import os
+import pathlib
+from typing import Iterator, Optional, Union
+
+import requests
+import tqdm
+
+from composer.core.time import Timestamp
+from composer.utils import dist
+from composer.utils.iter_helpers import iterate_with_pbar
+from composer.utils.object_store import ObjectStore
+
+__all__ = [
+    'GetFileNotFoundException',
+    'get_file',
+    'ensure_folder_is_empty',
+    'format_name_with_dist',
+    'format_name_with_dist_and_time',
+    'is_tar',
+]
+
+
+class GetFileNotFoundException(RuntimeError):
+    """Exception if :meth:`get_file` failed due to a not found error."""
+    pass
+
+
+def is_tar(name: Union[str, pathlib.Path]) -> bool:
+    """Returns whether ``name`` has a tar-like extension.
+
+    Args:
+        name (str | pathlib.Path): The name to check.
+
+    Returns:
+        bool: Whether ``name`` is a tarball.
+    """
+    return any(str(name).endswith(x) for x in (".tar", ".tgz", ".tar.gz", ".tar.bz2", ".tar.lzma"))
+
+
+def ensure_folder_is_empty(folder_name: Union[str, pathlib.Path]):
+    """Ensure that ``folder_name`` is empty.
```

Comment:
This is super nitpicky, but this wording makes me think that the literal variable `folder_name` (i.e. the string) is being checked for empty-ness, not the folder itself. I think it's clear enough just to say something like "Ensure that the given folder is empty."